lib/active_set/filtering/operation.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+require_relative '../attribute_instruction'
+
+class ActiveSet
+  module Filtering
+    class Operation
+      def initialize(set, instructions_hash)
+        @set = set
+        @instructions_hash = instructions_hash
```

Comment:
A valid `instructions_hash` would look like:
```
{
  AttributeInstructionString: Value
  String: {
    AttributeInstructionString: Value
  }
}
```

An `AttributeInstructionString` allows for a dot-separated string as an attribute path (e.g. `some.path.to.attribute`) as well as an operator suffix wrapped in parens (e.g. `some.path.to.attribute(>)`). Nested hashes will be resolved to a dot-separated AttributeInstructionString.